src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -228,18 +228,18 @@ private void validateOutputSchema() throws IOException {
     FieldList fields = bgSchema.getFields();
     // Match output schema field type with bigquery column type
     for (Schema.Field field : config.getSchema().getFields()) {
-      validateSimpleTypes(field);
+      validateSupportedTypes(field);
       BigQueryUtil.validateFieldSchemaMatches(fields.get(field.getName()), field, dataset, tableName);
     }
   }
 
-  private void validateSimpleTypes(Schema.Field field) {
+  private void validateSupportedTypes(Schema.Field field) {
     String name = field.getName();
     Schema fieldSchema = BigQueryUtil.getNonNullableSchema(field.getSchema());
     Schema.Type type = fieldSchema.getType();
 
-    // Complex types like arrays, maps and unions are not supported in BigQuery plugins.
-    if (!type.isSimpleType()) {
+    // Complex types like maps and unions are not supported in BigQuery plugins.
+    if ((!type.isSimpleType() && type != Schema.Type.ARRAY)) {
```

Comment:
ok I see this is where the deploy time validation is.

There needs to be a bit more checking done, checking that the component schema of the array is also valid.

Reply:
Since this is BQ source, we do not expect invalid arrays coming in from BQInputFormat.

Reply:
I thought the source lets the user modify the schema or use a macro?